iwyu_output.cc

```diff
@@ -72,53 +72,60 @@ class OutputLine {
   explicit OutputLine(const string& line)
       : line_(line) {}
   OutputLine(const string& line, const vector<string>& symbols)
-      : line_(line), symbols_(symbols) {}
+      : line_(line),
+        symbols_(symbols) {
+    symbols_.erase(std::remove_if(symbols_.begin(), symbols_.end(),
+                                  [](const string& v) { return v.empty(); }),
+                   symbols_.end());
+  }
 
   size_t line_length() const { return line_.size(); }
   bool needs_alignment() const { return !symbols_.empty(); }
   void add_prefix(const string& prefix) { line_ = prefix + line_; }
-  string printable_line(size_t desired_length, size_t max_length) const;
+  string printable_line(size_t min_length, size_t max_length) const;
 
  private:
   string line_;                     // '#include XXX' or 'class YYY;'
   vector<string> symbols_;          // symbols used from included header
 };
 
-string OutputLine::printable_line(size_t desired_length, size_t max_length)
-    const {
-  // Reduce max_length to make room for the ", etc" suffix.
-  max_length -= strlen(", etc");
-
-  desired_length = std::min(desired_length, max_length);
-
-  string retval = line_;
-
-  string prefix;   // what we print before each symbol in the 'why' comments
-  // We try to get the columns to line up nicely.
-  if (retval.length() < desired_length)
-    prefix += string(desired_length - retval.length(), ' ');
-  prefix += "  // for ";    // before 1st symbol, print ' // for '
-  int symbols_printed = 0;
-
-  for (Each<string> it(&symbols_); !it.AtEnd(); ++it) {
-    if (it->empty())       // ignore the empty ("") symbol
-      continue;
-    // At verbose levels of 0, 1, or 2, cut off output at max_length columns.
-    // Actually, at 74, to leave 5 chars for ', etc' and 1 for newline.
-    if (ShouldPrint(3) ||
-        retval.length() + prefix.length() + it->length() <= max_length) {
-      retval += prefix + *it;
-      ++symbols_printed;
-      prefix = ", ";       // before 2nd and subsequent symbols, print ', '
-    } else {
-      // Truncate at max_length cols.
-      if (symbols_printed > 0)
-        retval += ", etc";
-      break;
+// Append a helpful 'why' comment to the include line, containing the symbols
+// used from the header. Align nicely at lower verbosity levels.
+string OutputLine::printable_line(size_t min_length, size_t max_length) const {
```

Comment:
`max_length > 0` is required for correctness, so I added a `CHECK_` for that.

I think it's legal for `min_length` to be larger than `max_length`, however, e.g. if an `#include` line exceeds the configured max length, and that should be handled by the algorithm as-is.